app/vmselect/promql/aggr.go

```diff
@@ -649,15 +649,16 @@ func newAggrFuncTopK(isReverse bool) aggrFunc {
 		if err != nil {
 			return nil, err
 		}
+		lt := lessWithNaNs
+		if isReverse {
+			lt = lessWithNaNsReversed
+		}
```

Comment:
I think it would be more transparent, if we'll just filter out all NaNs before applying sorting. wdyt?